rotkehlchen/rotkehlchen.py

```diff
@@ -903,3 +910,44 @@ def create_oracle_cache(
             return  # only for cryptocompare for now
 
         self.cryptocompare.create_cache(from_asset, to_asset, purge_old)
+
+    def query_missing_prices(
+        self,
+        filter_query: HistoryEventFilterQuery,
+    ) -> None:
+        """Queries missing prices for HistoryBaseEntry in database based
+        on the query filter used"""
+        # Use a deepcopy to avoid mutations in the filter query if it is used later
+        new_filter_query = copy.deepcopy(filter_query)
```

Comment:
Yes, this filter has pagination and is later used in the logic (`rest.py`) to query the filtered events. What I do is update missing prices for events that will be returned

Reply:
Aha I see, so you don't query prices for all events but only for this page's events? Hmmm this will end up being weird at page switch. We really need to have a task going through all of them and getting prices if possible.

Essentially this: https://github.com/rotki/rotki/issues/3865